staging/src/k8s.io/api/resource/v1alpha3/types.go

```diff
@@ -0,0 +1,1010 @@
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
+
+	*RequestDetail `json:",inline" protobuf:"bytes,2,name=requestDetail"`
+}
+
+// RequestDetail is embedded inside Request. Exactly one field must be set.
+type RequestDetail struct {
+	// Device requests one or more devices.
+	//
+	// +optional
+	Device *DeviceRequest `json:"device,omitempty" protobuf:"bytes,1,opt,name=device"`
+}
+
+// DeviceRequest is currently the only permitted alternative in RequestDetail.
+type DeviceRequest struct {
+	// DeviceClassName references a specific DeviceClass, which can define
+	// additional configuration and selectors to be inherited by this
+	// request.
+	//
+	// A class is required. Which classes are available depends on the cluster.
+	//
+	// Administrators may use this to restrict which devices may get
+	// requested by only installing classes with selectors for permitted
+	// devices. If users are free to request anything without restrictions,
+	// then administrators can create an empty DeviceClass for users
+	// to reference.
+	//
+	// +required
+	DeviceClassName string `json:"deviceClassName" protobuf:"bytes,1,name=deviceClassName"`
+
+	// Selectors define criteria which must be satisfied by a specific
+	// device in order for that device to be considered for this
+	// request. All selectors must be satisfied for a device to be
+	// considered.
+	//
+	// +optional
+	// +listType=atomic
+	Selectors []Selector `json:"selectors,omitempty" protobuf:"bytes,2,name=selectors"`
+
+	// CountMode and its related fields define how many devices are needed
+	// to satisfy this request. Supported values are:
+	//
+	// - All: This request is for all of the matching devices in a pool.
+	//   Allocation will fail if some devices are already allocated,
+	//   unless adminAccess is requested.
+	//
+	// - Exact: This request is for a specific number of devices. The exact
+	//   number is provided in the count field.
+	//
+	// If countMode is not specified, the default countMode is Exact. If
+	// countMode is Exact and count is not specified, the default count is
+	// one. Any other requests must specify this field.
+	//
+	// More modes may get added in the future. Clients must refuse to handle
+	// requests with unknown modes.
+	//
+	// +default="Exact"
+	CountMode CountMode `json:"countMode,omitempty" protobuf:"bytes,3,opt,name=countMode"`
+
+	// Count is used only when the count mode is "Exact". Must be greater than zero.
+	// If CountMode is Exact and this field is not specified, the default is one.
+	//
+	// +optional
+	Count int64 `json:"count,omitempty" protobuf:"bytes,4,opt,name=count"`
+
+	// AdminAccess indicates that this is a claim for administrative access
+	// to the device(s). Claims with AdminAccess are expected to be used for
+	// monitoring or other management services for a device.  They ignore
+	// all ordinary claims to the device with respect to access modes and
+	// any resource allocations.
+	//
+	// Default is false.
+	//
+	// +optional
```

Comment:
+default=false

Reply:
Ack. Also removed the "Default is false.". At least for now, the `+default=false` is part of the doc text.